### PR TITLE
Refactor chunking methods

### LINKED ISSUE #44: Refactor WTS_Chunking
There are some irrelevant functions in the `WTS_chunking.py` file and `dynamic_yan_chunking` requires a docstring. 

Questions on whether to keep these annotation post-processing PyHa functions in this repository or plan to import them from PyHa-Lite. While keeping them in this repository would be nice for versioning, they will eventually be merged into PyHa-Lite AND `generate_automated_labels` is still required from PyHa.

Additionally, to make this more modular, read the .csvs in `__main__` and operate on the DataFrame only.

### LINKED ISSUE #52: Update all os.path to use pathlib
- chunking_methods
- dataset.py

### PR DESCRIPTION
**Pull request by sprestrelski**
Closes #44 
- update chunking methods to use DataFrames
- remove unused functions in WTS_Chunking
- use pathlib instead of os.path

### CODE REVIEW

File: chunking_methods/gen_csv_labels.py [hunk 1]
```diff
@@ -76,19 +75,17 @@ def generate_labels(path):
...
     for each chunk and reformat the columns for the training pipeline.
     Args:
-        metadata (str)
-            - Path to .csv with original audio clip information. Assumes 
+        metadata (DataFrame)
+            - DataFrame with original audio clip information. Assumes 
             Xeno-canto formatting.
-        strong_labels (str)
-            - Path to .csv with time-specific labels. Assumes PyHa formatting.  
+        strong_labels (DataFrame)
```
Comment (benjamin-cates) [inline chunking_methods/gen_csv_labels.py]: uhhh changed the argument name so this comment is wrong now

File: chunking_methods/gen_csv_labels.py [hunk 2]
```diff
@@ -182,26 +176,31 @@ def generate_raw_chunks(path, metadata, chunk_duration=5, filetype=".wav"):
...
-        generate_labels(CONFIG.data_path).to_csv(CONFIG.strong_labels)
-        attach_labels(
-            metadata=CONFIG.metadata,
-            strong_labels=CONFIG.strong_labels).to_csv(CONFIG.strong_labels)
+        labels = generate_labels(CONFIG.data_path, filetype=CONFIG.filetype)
+        labels.to_csv(CONFIG.strong_labels)
+
+        print("attaching strong labels...")
+        strong_labels = attach_labels(metadata, labels)
+        strong_labels.to_csv(CONFIG.strong_labels)
```
Comment (benjamin-cates) [inline chunking_methods/gen_csv_labels.py:194]: Thank you for making this code much more clear. It was really confusing earlier

**Comment by Sean1572** [comment]
Attempting to handle config issues

**Review by benjamin-cates — APPROVED** [review]
yay

**Pull request merged**